design/one-pager-automatic-dependency-downgrades.md

```diff
@@ -0,0 +1,76 @@
+# Automatic Dependency Version Downgrades
+
+* Owner: Ezgi Demirel Karabag (@ezgidemirel)
+* Reviewers: Crossplane Maintainers
+* Status: Accepted
+
+## Background
+
+Crossplane's package manager has traditionally handled missing dependencies by installing the latest valid versions 
+available using the lock Kubernetes object and a controller reconciles on it. This straightforward approach allowed for 
+quick setup but could not handle dynamic environments where dependency version change is required. In cases where the 
+dependency was installed but the constraints were no longer satisfied, the lock reconciler gives no error and package 
+revision reconciler notices the invalid version issue and returns an error to the user for manual intervention.
+
+With recent updates, the package manager has gained the capability to upgrade dependencies to the minimum valid version 
+rather than the latest, reducing the likelihood of breaking changes in existing environments. If no version meets the 
+constraints, the lock reconciler now displays the error message in the lock object’s status, complementing the error 
+messages from the package revision reconciler.
+
+It was deliberately decided not to downgrade the dependencies automatically to prevent unexpected behaviours in the 
+existing environments. However, this approach has its own drawbacks. The user has to manually intervene to downgrade the
+dependencies, which is not a good user experience and is not suitable for GitOps workflows. This proposal aims to 
+address this issue by introducing automatic dependency version downgrades behind a feature flag.
+
+![Package Manager diagram][pkg-mgr-diagram]
+
+## Possible Issues
+
+During dependency version downgrades, the following issues may occur:
+
+1) Missing CRDs
+2) Missing CRD fields
+3) CRD storage version changes
+
+Please note that these issues may also occur during package upgrades. But in that case, we assume the user is aware of 
+the changes and accepts the risks.
```

Comment:
> we assume the user is aware of the changes and accepts the risks.

Provider authors may also be more careful about upgrades but might have overlooked for downgrade cases with a higher probability.